### PR TITLE
NPD4404 12.5A AOne Inline Relay.

### PR DESCRIPTION
**Pull request by Zohair-Hussain-Aurora**
NPD4404 12.5A AOne Inline Relay version 1 firmware.

### CODE REVIEW

File: NPD4404_12_5AInlineRelay_callbacks.c [hunk 1]
```diff
@@ -68,11 +66,9 @@ bool emberAfMainStartCallback(int* returnCode,
                               char** argv)
 {
    MCUInit(COM_USART1);
-
-   // The soonest we can put the light on, we do
    if(!ResetWasBootloader())
    {
-      MCURequestLevelModeSingleColour(LIGHT_LEVEL_MAX_ZCL);
+	  MCURequestOnOffModeRelay(RELAY_OFF);
    }
```
Comment (iandaintith) [inline NPD4404_12_5AInlineRelay_callbacks.c]: Check indentation

File: NPD4404_12_5AInlineRelay_callbacks.c [hunk 2]
```diff
@@ -192,17 +183,7 @@ void emberAfOnOffClusterServerAttributeChangedCallback(uint8_t endpoint,
...
-}
-
-// Called when the Level control cluster has changed
-void emberAfLevelControlClusterServerAttributeChangedCallback(int8u endpoint,
-      EmberAfAttributeId attributeId)
-{
-   if (ZCL_CURRENT_LEVEL_ATTRIBUTE_ID == attributeId)
-   {
-      s_zigbeeChangePending = true;
+	  s_zigbeeChangePending = true;
```
Comment (iandaintith) [inline NPD4404_12_5AInlineRelay_callbacks.c]: indentation

File: NPD4404_12_5AInlineRelay_callbacks.c [hunk 3]
```diff
@@ -68,11 +66,9 @@ bool emberAfMainStartCallback(int* returnCode,
                               char** argv)
 {
    MCUInit(COM_USART1);
-
-   // The soonest we can put the light on, we do
    if(!ResetWasBootloader())
    {
-      MCURequestLevelModeSingleColour(LIGHT_LEVEL_MAX_ZCL);
+	  MCURequestOnOffModeRelay(RELAY_OFF);
```
Comment (iandaintith) [inline NPD4404_12_5AInlineRelay_callbacks.c]: check indentation

File: NPD4404_12_5AInlineRelay_callbacks.c [hunk 4]
```diff
@@ -136,17 +131,15 @@ void emberAfPluginConnectionManagerFinishedCallback(EmberStatus status)
...
-      ClusterSetOTAClient(LIGHT_ENDPOINT, EMBER_AF_PLUGIN_OTA_CLIENT_POLICY_FIRMWARE_VERSION,
-             EMBER_AF_MANUFACTURER_CODE , EMBER_AF_PLUGIN_OTA_CLIENT_POLICY_IMAGE_TYPE_ID);
+      ClusterSetOTAClient(RELAY_ENDPOINT, EMBER_AF_PLUGIN_OTA_CLIENT_POLICY_FIRMWARE_VERSION,
+          EMBER_AF_MANUFACTURER_CODE, EMBER_AF_PLUGIN_OTA_CLIENT_POLICY_IMAGE_TYPE_ID);
 
       if(s_indicateJoinSuccess)
       {
          s_indicateJoinSuccess = false;
          TRACE(TRACE_ZIGBEE, "ZIGBEE: Join OK\r\n");
          s_uiActive = true;
```
Comment (iandaintith) [inline NPD4404_12_5AInlineRelay_callbacks.c]: Do we have a UI concept in this product?  ie) is there an LED that the user sees?

File: NPD4404_12_5AInlineRelay_callbacks.c [hunk 5]
```diff
@@ -214,9 +195,14 @@ void emberAfPluginIdentifyStartFeedbackCallback(u8 endpoint,
 
    if(s_startupComplete)
    {
-      if(LIGHT_ENDPOINT == endpoint)
+      if(RELAY_ENDPOINT == endpoint)
       {
-         s_uiActive = true;
+         u8 onOff = ClusterReadOnOff(RELAY_ENDPOINT);
```
Comment (iandaintith) [inline NPD4404_12_5AInlineRelay_callbacks.c]: May as well remove the read and just set to  ClusterWriteOnOff(RELAY_ENDPOINT, TRUE)

**Comment by iandaintith** [comment]
I have fixes to build in SS4 and fixes such that identify doesn't change the values in the on/off cluster. 
But you havent added me with any permissions on this repo. You will have to add me in the morning.

File: NPD4404_12_5AInlineRelay_callbacks.c (hunk 4 above)
Comment (Zohair-Hussain-Aurora) [reply NPD4404_12_5AInlineRelay_callbacks.c]: inkedimg_20190214_092006_li
inkedimg_20190214_091919_li

Comment (Zohair-Hussain-Aurora) [reply NPD4404_12_5AInlineRelay_callbacks.c]: Yes. Just like the Fixed white strip.

**Comment by auroramikej** [comment]
Hi Ian, done.



Cheers

Mike





*Mike Jones*

Technology & IoT Director - Aurora Lighting Group



[image: New AU Logo PNG]



*Mobile*: +44 (0) 7392 080 722
*Skype*: aurora.mikej  |  http://auroralighting.com



[image: 2019-01_AOne™-GenAOne™-General--Static-2x]



Aurora Lighting, Unit A Frankland Road, Swindon, Wiltshire SN58YG























*From:* iandaintith <notifications@github.com>
*Sent:* 13 February 2019 20:33
*To:* AuroraLighting/NPD4404_12_5AInlineRelay <
NPD4404_12_5AInlineRelay@noreply.github.com>
*Cc:* Subscribed <subscribed@noreply.github.com>
*Subject:* Re: [AuroraLighting/NPD4404_12_5AInlineRelay] NPD4404 12.5A AOne
Inline Relay. (#1)



I have fixes to build in SS4 and fixes such that identify doesn't change
the values in the on/off cluster.
But you havent added me with any permissions on this repo. You will have to
add me in the morning.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_AuroraLighting_NPD4404-5F12-5F5AInlineRelay_pull_1-23issuecomment-2D463357406&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=THw0WDaG0DT7pdo_RurMPBYiHezCOiY3-DRC5PAUO0k&m=jpdKn01BnOpswIR58w_SCNGSJLq48CQ91q34X2PSmdw&s=I5u35_5i3FRx5OAXSEV3XFo8GavegO0vQs5QaEiaxP4&e=>,
or mute the thread
<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ARlkBhUoAx2gK4INBC4-5FOzMa0J8GR7RVks5vNHaHgaJpZM4a55oj&d=DwMFaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=THw0WDaG0DT7pdo_RurMPBYiHezCOiY3-DRC5PAUO0k&m=jpdKn01BnOpswIR58w_SCNGSJLq48CQ91q34X2PSmdw&s=dpB_uGiTjJ1kg3kNWI8cdKG-Plk9VvhAD4Qrh2btxc4&e=>
.


------------------------------

This email has been scanned for spam and viruses by Proofpoint Essentials.
Click here
<https://eu1.proofpointessentials.com/index01.php?mod_id=11&mod_option=logitem&mail_id=1550089993-Pb9CT5YIQcIN&r_address=mike.jones%40auroralighting.com&report=1>
to report this email as spam.

File: NPD4404_12_5AInlineRelay_callbacks.c (hunk 4 above)
Comment (iandaintith) [reply NPD4404_12_5AInlineRelay_callbacks.c]: OK. Is that controlled by Relay On/Off

**Pull request merged**